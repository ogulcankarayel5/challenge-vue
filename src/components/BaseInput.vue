<template>
  <div class="input">
    <label class="input-label" :for="`${label}`">{{ label }}</label>
    <input
      class="input-value"
      v-bind="$attrs"
      :value="modelValue"
      @input="handleInput"
      :id="`${label}`"
    />
  </div>
</template>

<script>
export default {
  name: "BaseInput",
  props: {
    modelValue: {
      type: String,
    },
    label: {
      type: String,
    },
  },
  methods: {
    handleInput(event) {
      this.$emit("update:modelValue", event.target.value);
    },
  },
};
</script>

<style lang="sass" scoped>
.input
  margin-top: 15px
  display: flex
  flex-direction: column
  &-label
    padding-left: 15px
  &-value
    padding: 10px 10px 10px 2px
    border: 1px solid #000
    border-radius: 3px
    width: 25rem
</style>
