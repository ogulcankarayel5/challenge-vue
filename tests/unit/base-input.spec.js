import { mount } from "@vue/test-utils";
import BaseInput from "@/components/BaseInput.vue";

const factory = (props = {}, slots = {}) => {
  return mount(BaseInput, {
    props: { ...props },
    slots: { ...slots },
  });
};

describe("BaseBox.vue", () => {
  it("snapshot", () => {
    const wrapper = factory();

    expect(wrapper.element).toMatchSnapshot();
  });
  it("changes value correctly", async () => {
    const wrapper = factory({ modelValue: "" });

    const input = wrapper.find(".input-value");
    expect(input.element.value).toBe("");
    await input.setValue("deneme");

    expect(input.element.value).toBe("deneme");
  });
  it("renders label name", () => {
    const wrapper = factory({ label: "user" });

    expect(wrapper.find(".input-label").text()).toBe("user")
  })
});
