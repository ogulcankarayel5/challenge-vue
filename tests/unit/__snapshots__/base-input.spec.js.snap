// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseBox.vue snapshot 1`] = `
<div
  class="input"
>
  <label
    class="input-label"
    for="undefined"
  />
  <input
    class="input-value"
    id="undefined"
  />
</div>
`;
